### ISSUE TITLE
Append number at end of default function name only if conflict

### ISSUE THREAD

**Issue by anthonychu** [microsoft/vscode-azurefunctions#2067]
Is it possible to append a number at the end of the default function name _only_ if there's a conflict? Would be great for the first function not to have a `1` appended.

I think in most cases, `HttpTrigger` is more desirable than `HttpTrigger1` when creating the first function.

There are also situations where the function name matters:
* In the Durable Functions quickstart, the durable activity function must be named `Hello`, not `Hello1`.
* The "negotiate" function in SignalR Service must be named `negotiate` (although this is addressed by hardcoding the route to `/negotiate` in the function.json template itself.

**Comment by EricJizbaMSFT** [comment]
That was the previous behavior and it was intentionally changed to always have a one. The short reason is that it makes clear it's an _instance_ of a trigger and it follows other patterns seen in products like VS. Reference https://github.com/microsoft/vscode-azurefunctions/issues/1787

For the durable/negotiate templates, sounds like we might need more discussion there. What's the point of prompting for a name if it can't be changed? On the flip side - what's the point of having a template when I can't even specify the name? Would be great if we could figure out a way to make that way less fragile

**Comment by davidmrdavid** [comment]
Hey @EricJizbaMSFT , just following-up on this thread since the topic came up during some testing today. What do you mean by "What's the point of prompting for a name if it can't be changed? On the flip side - what's the point of having a template when I can't even specify the name?"

In any case, another alternative would be to give us the possibility of using the Functions plug-in to incorporate fully realized *samples* in addition to individual functions, right? That way, people could immediately play with a well-integrated set of functions without worrying about renaming **and** getting all the necessary functions. 

Can we make this happen?

**Comment by EricJizbaMSFT** [comment]
@anthonychu said "the durable activity function must be named `Hello`". I assume that's because the durable _orchestration_ function hard-codes "Hello" and they need to be the same. My point was that it's not a very useful template if the name has to be the same thing every time. That means users will run into problems as soon as they try to create a second durable function in a project. In addition, the durable templates are fragile because users have to know to create all 3 functions together and keep the name "Hello" consistent between the activity and orchestration functions. Ideally there's one template that creates all 3 functions, replacing "Hello" in both the activity and orchestration functions with whatever name the user chooses.

We already have some support for samples. They're hosted [here](https://aka.ms/AA4ul9b) and we have a link when users first create a project:
<img width="414" alt="Screen Shot 2020-06-22 at 2 00 50 PM" src="https://user-images.githubusercontent.com/11282622/85334913-e9dbae00-b490-11ea-93d7-e2b00de0da49.png">

**Comment by anthonychu** [comment]
There are a couple of related issues here. The first is the whether we should append a number to the default name even if there is no conflict. IMHO it's the default name, so it should be the default unless there is already a function with that name.

If the concern is that it's not clear that it's an instance of a function, we can address that in the [template metadata](https://github.com/Azure/azure-functions-templates/blob/dev/Functions.Templates/Templates/HttpTrigger-Python/metadata.json#L2) by changing the default name to something that makes it clear they are instances. For instance, we should default HttpTrigger function names to `HttpExample` as we're using that in the [quickstarts](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-function-vs-code?pivots=programming-language-python#create-an-azure-functions-project) anyway.

The simplest Durable Functions project requires 3 functions. Ideally there's a way to create complex templates that consist of multiple functions. This needs more discussion, maybe in the templates repo.

**Comment by davidmrdavid** [comment]
Yeah I'm not advocating for the name not to change from `Hello`. All I'm saying is that the first function created could be `Hello` and the second one could be `Hello2` and then proceed with the numbering, no?

And yeah, I think we all agree that ideally we can create templates consisting of **multiple functions at once.** That would be fantastic! In my opinion, this would be preferrable to worrying about whether or not the function has a number after its name or not. So long as we can get multi-function templates that "just work", then I'm satisfied. Like Anthony said, I think we want an experience where the plugin is enough to get you running with the simplest Durable Functions use-case, and it currently does **not** do that without manual edits that complete beginners are unlikely to be aware of.
